### ISSUE TITLE
Could not load file or assembly 'Microsoft.CodeAnalysis, Version=4.5.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. The system cannot find the file specified.

### ISSUE THREAD

**Issue by reza-toorani** [dotnet/format#1805]
We are facing this error 
```
 Could not load file or assembly 'Microsoft.CodeAnalysis, Version=4.5.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. The system cannot find the file specified.
   at System.Reflection.RuntimeModule.GetTypes(RuntimeModule module)
   at System.Reflection.Assembly.GetTypes()
   at Microsoft.CodeAnalysis.Tools.Analyzers.AnalyzerFinderHelpers.<>c.<LoadAnalyzersAndFixers>b__0_0(Assembly assembly)
   at System.Linq.Enumerable.SelectManySingleSelectorIterator`2.MoveNext()
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at System.Linq.Enumerable.OfTypeIterator[TResult](IEnumerable source)+MoveNext()
   at System.Collections.Generic.LargeArrayBuilder`1.AddRange(IEnumerable`1 items)
   at System.Collections.Generic.EnumerableHelpers.ToArray[T](IEnumerable`1 source)
   at System.Collections.Immutable.ImmutableArray.CreateRange[T](IEnumerable`1 items)
   at Microsoft.CodeAnalysis.Tools.Analyzers.AnalyzerFinderHelpers.LoadAnalyzersAndFixers(IEnumerable`1 assemblies)
   at Microsoft.CodeAnalysis.Tools.Analyzers.AnalyzerReferenceInformationProvider.GetAnalyzersAndFixers(Project project)
   at System.Collections.Immutable.ImmutableDictionary.<>c__DisplayClass9_0`3.<ToImmutableDictionary>b__0(TSource element)
   at System.Linq.Utilities.<>c__DisplayClass2_0`3.<CombineSelectors>b__0(TSource x)
   at System.Linq.Enumerable.SelectIListIterator`2.MoveNext()
   at System.Collections.Immutable.ImmutableDictionary`2.AddRange(IEnumerable`1 items, MutationInput origin, KeyCollisionBehavior collisionBehavior)
   at System.Collections.Immutable.ImmutableDictionary`2.AddRange(IEnumerable`1 pairs, Boolean avoidToHashMap)
   at System.Collections.Immutable.ImmutableDictionary`2.AddRange(IEnumerable`1 pairs)
   at System.Collections.Immutable.ImmutableDictionary.ToImmutableDictionary[TSource,TKey,TValue](IEnumerable`1 source, Func`2 keySelector, Func`2 elementSelector, IEqualityComparer`1 keyComparer, IEqualityComparer`1 valueComparer)
   at System.Collections.Immutable.ImmutableDictionary.ToImmutableDictionary[TSource,TKey,TValue](IEnumerable`1 source, Func`2 keySelector, Func`2 elementSelector)
   at Microsoft.CodeAnalysis.Tools.Analyzers.AnalyzerReferenceInformationProvider.GetAnalyzersAndFixers(Solution solution, FormatOptions formatOptions, ILogger logger)
   at Microsoft.CodeAnalysis.Tools.Analyzers.AnalyzerFormatter.FormatAsync(Solution solution, ImmutableArray`1 formattableDocuments, FormatOptions formatOptions, ILogger logger, List`1 formattedFiles, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.Tools.CodeFormatter.RunCodeFormattersAsync(Solution solution, ImmutableArray`1 formattableDocuments, FormatOptions formatOptions, ILogger logger, List`1 formattedFiles, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.Tools.CodeFormatter.FormatWorkspaceAsync(FormatOptions formatOptions, ILogger logger, CancellationToken cancellationToken, String binaryLogPath)
   at Microsoft.CodeAnalysis.Tools.FormatCommandCommon.FormatAsync(FormatOptions formatOptions, ILogger`1 logger, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.Tools.Commands.RootFormatCommand.FormatCommandDefaultHandler.InvokeAsync(InvocationContext context)
   at System.CommandLine.Invocation.InvocationPipeline.<>c__DisplayClass4_0.<<BuildInvocationChain>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.CommandLineBuilderExtensions.<>c__DisplayClass16_0.<<UseParseErrorReporting>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.CommandLineBuilderExtensions.<>c__DisplayClass11_0.<<UseHelp>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.CommandLineBuilderExtensions.<>c__DisplayClass21_0.<<UseVersionOption>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.CommandLineBuilderExtensions.<>c__DisplayClass18_0.<<UseTypoCorrections>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.CommandLineBuilderExtensions.<>c.<<UseSuggestDirective>b__17_0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.CommandLineBuilderExtensions.<>c__DisplayClass15_0.<<UseParseDirective>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.CommandLineBuilderExtensions.<>c.<<RegisterWithDotnetSuggest>b__4_0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.CommandLineBuilderExtensions.<>c__DisplayClass7_0.<<UseExceptionHandler>b__0>d.MoveNext() 
   ```
This is the project specification.
```
   <Project Sdk="Microsoft.NET.Sdk">
    <PropertyGroup>
      <TargetFramework>net6.0</TargetFramework>
      <ImplicitUsings>enable</ImplicitUsings>
     <Nullable>enable</Nullable>
     <EnforceCodeStyleInBuild>true</EnforceCodeStyleInBuild>
    <IsPackable>false</IsPackable>
  </PropertyGroup>
 </Project>

 ```
And here is the pipeline detail
```
 - task: UseDotNet@2
            displayName: Install .NET 6.0 SDK
            inputs:
              packageType: 'sdk'
              version: 6.0.x
      - task: CmdLine@2
            displayName: Verify code formatting
            inputs:
              script: |
                dotnet format ./src/Project.sln  --verify-no-changes --severity error --no-restore
              failOnStderr: true
```

 The dotnet runtime version is '7.0.3'.
 The dotnet CLI version is '7.0.201'.

Everything was fine before we upgraded dotnet to version 7.0. While getting the error locally, the pipeline fails as well.

Can the reason be the same as https://github.com/dotnet/format/issues/1800?

**Comment by koenigst** [comment]
We have the same issue after migrating from dotnet SDK `7.0.102` to `7.0.201`.

**Comment by reza-toorani** [comment]
To fix this issue we set false `EnforceCodeStyleInBuild`

**Comment by lbruun** [comment]
Hit by this too.

For GitHub Workflow we had to do the following in order to lock to a previous SDK version:

1. In the workflow file:

```yaml
  - name: Setup .NET SDK
    uses: actions/setup-dotnet@v3
    with:
      dotnet-version: '7.0.103'
```

2. Add a `global.json` file in the root of the repo with this content:

```json
{
    "sdk": {
      "version": "7.0.103"
    }
}
```

The first step merely makes sure the version is available on the Runner. The second step is necessary to make sure the CLI actually picks up the SDK version we want.

**Comment by herebebeasties** [comment]
Please close this as a duplicate of #1800.

**Comment by MarkusRodler** [comment]
I have the same error but I think it is the same one as last year: https://github.com/dotnet/format/issues/1519
Infos:
- `EnforceCodeStyleInBuild` is not set
- I am using `dotnet format`
- The dotnet runtime version is '7.0.4'
- The dotnet CLI version is '7.0.202'
- I have a library-project including `Microsoft.CodeAnalysis.CSharp 4.5.0`. If I lower it to `4.4.0` than there is no error.

**Comment by nietras** [comment]
Same issue. Multiple machines incl. in CI. Tried to downgrade `global.json` to `7.0.102` from `7.0.202`. Doesn't work still. Don't have a working fix neither local or on CI. Have to disable the task for CI. This issue has hit us several times before.

**Comment by othmane-kinane-nw** [comment]
> Same issue. Multiple machines incl. in CI. Tried to downgrade `global.json` to `7.0.102` from `7.0.202`. Doesn't work still. Don't have a working fix neither local or on CI. Have to disable the task for CI. This issue has hit us several times before.

@nietras I did manage to make it work on CI with SDK 7.0.104 by setting DOTNET_INSTALL_DIR and DOTNET_ROOT, so that the SDK used in the workflow is completely isolated from the one installed on the agent:

```yaml
name: .NET

on:
  push:
    branches: [ main ]
  pull_request:
    branches: [ main ]
  workflow_dispatch:

env:
  PROJECT_DIR: ${{ github.workspace }}/backend
  DOTNET_CLI_TELEMETRY_OPTOUT: true
  CONFIGURATION: Release
  DOTNET_INSTALL_DIR: .temp/dotnet
  DOTNET_ROOT: .temp/dotnet

jobs:
  build:
    name: Build and test
    runs-on: ubuntu-latest
    defaults:
      run:
        shell: bash
        working-directory: ${{ env.PROJECT_DIR }}
    steps:    
    - uses: actions/checkout@v3    
    - name: Setup .NET
      uses: actions/setup-dotnet@v3
      with:
        global-json-file: backend/global.json        
    - name: Restore dependencies
      run: dotnet restore      
    - name: Check format
      run: dotnet format --no-restore --verify-no-changes --verbosity detailed
```

Here is the content of my global.json file:
```json
{
  "sdk": {
    "version": "7.0.104",
    "rollForward": "latestMinor"
  }
}
```

**Comment by GGG-KILLER** [comment]
I've been getting this using .NET 8 in a dev container where .NET was installed using `dotnet-install.sh`.

The project it runs into an exception while trying to format is an Incremental Source Generator.

Solution: https://github.com/LorettaDevs/Loretta (devcontainer files are contained within)
Project: `Loretta.Generators.SyntaxFactsGenerator`
Format command: `dotnet format` (v8.0.405501+2cb3e68c6b9a966114572fd63f2a20d2cb54a288)
Output: https://gist.github.com/GGG-KILLER/81a9773f7c02a12a841edd331751d2ee


<details>
<summary><code>dotnet --info</code></summary>

```console
$ dotnet --info
.NET SDK:
 Version:   8.0.100-preview.2.23157.25
 Commit:    54801b2435

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  22.04
 OS Platform: Linux
 RID:         ubuntu.22.04-x64
 Base Path:   /usr/local/dotnet/current/sdk/8.0.100-preview.2.23157.25/

Host:
  Version:      8.0.0-preview.2.23128.3
  Architecture: x64
  Commit:       30b879924a

.NET SDKs installed:
  6.0.407 [/usr/local/dotnet/current/sdk]
  7.0.202 [/usr/local/dotnet/current/sdk]
  8.0.100-preview.2.23157.25 [/usr/local/dotnet/current/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.15 [/usr/local/dotnet/current/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.4 [/usr/local/dotnet/current/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 8.0.0-preview.2.23153.2 [/usr/local/dotnet/current/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.15 [/usr/local/dotnet/current/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.4 [/usr/local/dotnet/current/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 8.0.0-preview.2.23128.3 [/usr/local/dotnet/current/shared/Microsoft.NETCore.App]

Other architectures found:
  None

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download
```
</details>

**Comment by lbruun** [comment]
> Please close this as a duplicate of #1800.

@herebebeasties. Not sure it is the same. This ticket is for folks who has _not_ set `EnforceCodeStyleInBuild`. In #1800 people report that they can avoid the problem by commenting out `EnforceCodeStyleInBuild`. 

It is more likely a re-occurrence of #1519.

**Comment by glen-84** [comment]
@reza-toorani Is this fixed for you in 7.0.304? If so, please close this issue.
